File: frontend/services/video.ts
/**
 * Video token client.
 *
 * The browser never holds LiveKit credentials — it asks the backend for a token scoped to
 * one room and one identity, and that token is all it ever sees.
 */

import { apiRequest } from "@/services/api-client";

export interface RoomToken {
  /** LiveKit server URL (`wss://…`). */
  url: string;
  /** Short-lived JWT granting join access to this debate. */
  token: string;
  roomName: string;
}

export async function getRoomToken(roomId: string, authToken: string | null): Promise<RoomToken> {
  return apiRequest<RoomToken>(`/rooms/${roomId}/token`, { method: "POST", token: authToken });
}

export const videoKeys = {
  token: (roomId: string) => ["video", "token", roomId] as const,
};

File: backend/app/schemas/video.py
"""Video token response models."""

from app.schemas.base import CamelModel


class RoomToken(CamelModel):
    """Everything the browser needs to join the debate's media room.

    Deliberately does not include the API key: the token is already scoped and signed, and
    the client has no use for the credentials that produced it.
    """

    url: str
    """The LiveKit server URL (``wss://…``). Public."""

    token: str
    """Short-lived JWT granting join access to this room as this participant."""

    room_name: str
